lib/model/folder_sendrecv.go

```diff
@@ -1768,6 +1786,13 @@ loop:
 				lastFile = job.file
 			}
 
+			if !job.file.IsDeleted() && !job.file.IsInvalid() {
+				// Now that the file is finalized, grab possibly updated
+				// info from disk into the local FileInfo.
+				if err := f.updateFileInfoMetadata(&job.file); err != nil {
```

Comment:
The only thing we update here is the inode change time, which we need for comparisons in the scanner. We need to do it here because it's affected by things like chmod, chtimes and setting xattrs which we've done previously. That attribute is never sent over the wire so there are no discrepancies.

For the actual xattrs, we don't change what's in the global version based on our filtering etc when we apply. However, if the file is then changed locally in whatever way, our filtering will be applied when we scan it. I think this is as desired...

Reply:
Right. Two suggestions to make that more understandable (to me at least :) ):

Add a comment, e.g. "Operations while pulling might have changed inode-time. The scanner uses that to detect changes. Thus the file info is update such that the subsequent scan doesn't pick it up."

Metadata is just inode-change set. xattrs, owner, group, perms, ... are something else than metadata. That's not obvious to me at least. I'd be in favor of renaming it to e.g. `fileInfoUpdateInodeTime`. If there is more metadata in the future, we can always create a generic function again and call the inode-time one from there.